lib/services/local/assistant.go

```diff
@@ -0,0 +1,204 @@
+/*
+ *
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package local
+
+import (
+	"context"
+	"encoding/json"
+	"sort"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/lib/backend"
+)
+
+// Conversation is a conversation entry in the backend.
+type Conversation struct {
+	Title          string    `json:"title,omitempty"`
+	ConversationID string    `json:"conversation_id"`
+	CreatedTime    time.Time `json:"created_time"`
+}
+
+// CreateAssistantConversation creates a new conversation entry in the backend.
+func (s *IdentityService) CreateAssistantConversation(ctx context.Context, username string,
+	req *proto.CreateAssistantConversationRequest,
+) (*proto.CreateAssistantConversationResponse, error) {
+	if username == "" {
+		return nil, trace.BadParameter("missing parameter username")
+	}
+
+	conversationID := uuid.New().String()
+	payload := &Conversation{
+		ConversationID: conversationID,
+		CreatedTime:    req.CreatedTime,
+	}
+
+	value, err := json.Marshal(payload)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	item := backend.Item{
+		Key:   backend.Key(assistantConversationPrefix, username, conversationID),
+		Value: value,
+	}
+
+	_, err = s.Create(ctx, item)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &proto.CreateAssistantConversationResponse{Id: conversationID}, nil
+}
+
+func (s *IdentityService) getConversation(ctx context.Context, username, conversationID string) (*Conversation, error) {
+	item, err := s.Get(ctx, backend.Key(assistantConversationPrefix, username, conversationID))
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	var conversation Conversation
+	if err := json.Unmarshal(item.Value, &conversation); err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &conversation, nil
+}
+
+// SetAssistantConversationTitle sets the given title as the assistant conversation title.
+func (s *IdentityService) SetAssistantConversationTitle(ctx context.Context, username string, request *proto.ConversationInfo) error {
```

Comment:
Do you anticipate having to provide ability to update any other conversation properties in future?

If so, would it make more sense to have a more generic `UpdateAssistantConversation` method which would only allow updating title for now but can be extended in future?

Reply:
I didn't want to change the API, but while addressing a different comment, I've also changed this.